spinalcordtoolbox/deepseg/models.py

```diff
@@ -357,3 +359,23 @@ def get_metadata(folder_model):
     with open(fname_metadata, "r") as fhandle:
         metadata = json.load(fhandle)
     return metadata
+
+
+def onnx_inference(model_path, input_data):
+    """
+    Perform inference using an '.onnx' model.
+
+    NB: This function is used by sct_deepseg_sc, sct_deepseg_gm, and sct_deepseg_lesion,
+        NOT sct_deepseg. It's possible that the modules and submodules could be organized
+        differently to better indicate the distinctions between old/new deepseg functions.
+    """
+    # This option helps to combat an issue where the CPU memory arena would unnecessarily
+    # consume ~6 gigabytes of memory, when our models only truly use ~5 megabytes. See also:
+    # https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3738#discussion_r881735426
+    sess_options = ort.SessionOptions()
+    sess_options.enable_cpu_mem_arena = False
+
+    ort_sess = ort.InferenceSession(model_path, sess_options=sess_options)
+    preds = ort_sess.run(output_names=["predictions"], input_feed={"input_1": input_data})
+
+    return preds
```

Comment:
It's possible that there's a better place to put this function?

Currently, we have the following files in the `spinalcordtoolbox/` package:

* `deepseg/models.py`: The "new" deepseg (ivadomed Torch-based models)
* `deegseg_gm/deepseg_gm.py`: The "old" deepseg (`sct_deepseg_gm`)
* `deepseg_sc/core.py`: The "old" deepseg (`sct_deepseg_sc`)
* `deepseg_lesion/core.py`: The "old" deepseg (`sct_deepseg_lesion`)

So, there wasn't really a great option for where to put `onnx_inference()`, since it's used by all 3 of the "old" deepseg functions.

Reply:
Did you planned to remove the "old" deepseg one day to be completly replaced by the "new" deepseg ?
If yes I would say that it's better to not put it in the "new" deepseg so in the future you don't have cross-dependency and it will be more easy to remove the "old" one.

Maybe this function could feat a new shared/common folder, so if you have other common scenario it will not be a question anymore

Reply:
> Did you planned to remove the "old" deepseg one day to be completly replaced by the "new" deepseg ? If yes I would say that it's better to not put it in the "new" deepseg so in the future you don't have cross-dependency and it will be more easy to remove the "old" one.

That would be the ideal? (See: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3046)

But, realistically speaking, the work required to do this is significant, and likely beyond my capacities as a solo developer. To do this would require a dedicated student training replacement models 

> Maybe this function could feat a new shared/common folder, so if you have other common scenario it will not be a question anymore

Still, I think you make a good point about module organization! I would definitely be willing to try merging the `deepseg_x` functions together into a single module folder.

I've done this in [42c87d8](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3738/commits/42c87d8989b3fb746541fe84b2c85619256fad21) and [80044ec](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3738/commits/80044eca36d7bb04526c8710b3d5034e4e34562c). :)

Reply:
![image](https://user-images.githubusercontent.com/16181459/173638605-4f44ed58-f4d9-47dc-8f93-36d60c2b128c.png)

I've done this in [42c87d8](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3738/commits/42c87d8989b3fb746541fe84b2c85619256fad21) and [80044ec](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3738/commits/80044eca36d7bb04526c8710b3d5034e4e34562c). :)